# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsCookieService.cpp
# Commit: f2df73fda468
# Full Hash: f2df73fda46859150605f33bcdaeb49ab7037720
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-10-19 21:40:34
# Regressor Bug: 1588899
# File Overlap Count: 1
# Description:
#   Bug 1588899 - P1. Move classification flags related method to nsIClassifiedChannel. r=Ehsan,baku
#   
#   This is where it should have been in the first place. Those attributes belong there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49577
# ==============================================================================

diff -r b98968eb71d3 -r f2df73fda468 netwerk/cookie/nsCookieService.cpp
--- a/netwerk/cookie/nsCookieService.cpp	Thu Oct 17 15:24:31 2019 +0000
+++ b/netwerk/cookie/nsCookieService.cpp	Sat Oct 19 04:30:24 2019 +0000
@@ -19,6 +19,7 @@
 
 #include "nsCookieService.h"
 #include "nsContentUtils.h"
+#include "nsIClassifiedChannel.h"
 #include "nsIServiceManager.h"
 #include "nsIScriptSecurityManager.h"
 #include "nsIWebProgressListener.h"
@@ -2012,15 +2013,16 @@
   bool isTrackingResource = false;
   bool firstPartyStorageAccessGranted = false;
   uint32_t rejectedReason = 0;
-  nsCOMPtr<nsIHttpChannel> httpChannel = do_QueryInterface(aChannel);
-  if (httpChannel) {
-    isTrackingResource = httpChannel->IsTrackingResource();
+  nsCOMPtr<nsIClassifiedChannel> classifiedChannel =
+      do_QueryInterface(aChannel);
+  if (classifiedChannel) {
+    isTrackingResource = classifiedChannel->IsTrackingResource();
 
     // Check first-party storage access even for non-tracking resources, since
     // we will need the result when computing the access rights for the reject
     // foreign cookie behavior mode.
     if (AntiTrackingCommon::IsFirstPartyStorageAccessGrantedFor(
-            httpChannel, aHostURI, &rejectedReason)) {
+            aChannel, aHostURI, &rejectedReason)) {
       firstPartyStorageAccessGranted = true;
     }
   }
@@ -2134,15 +2136,16 @@
   bool isTrackingResource = false;
   bool firstPartyStorageAccessGranted = false;
   uint32_t rejectedReason = 0;
-  nsCOMPtr<nsIHttpChannel> httpChannel = do_QueryInterface(aChannel);
-  if (httpChannel) {
-    isTrackingResource = httpChannel->IsTrackingResource();
+  nsCOMPtr<nsIClassifiedChannel> classifiedChannel =
+      do_QueryInterface(aChannel);
+  if (classifiedChannel) {
+    isTrackingResource = classifiedChannel->IsTrackingResource();
 
     // Check first-party storage access even for non-tracking resources, since
     // we will need the result when computing the access rights for the reject
     // foreign cookie behavior mode.
     if (AntiTrackingCommon::IsFirstPartyStorageAccessGrantedFor(
-            httpChannel, aHostURI, &rejectedReason)) {
+            aChannel, aHostURI, &rejectedReason)) {
       firstPartyStorageAccessGranted = true;
     }
   }